sql/core/src/test/scala/org/apache/spark/sql/execution/datasources/xml/TestXmlData.scala

```diff
@@ -68,4 +68,444 @@ private[xml] trait TestXmlData {
+
+  def complexFieldAndType2: Seq[String] =
+    """
+          |<ROW>
+          |  <arrayOfStruct>
+          |    <item>
+          |      <field1>true</field1>
+          |      <field2>str1</field2>
+          |    </item>
+          |    <item>
+          |      <field1>false</field1>
+          |    </item>
+          |    <item>
+          |      <field3/>
+          |    </item>
+          |  </arrayOfStruct>
+          |  <complexArrayOfStruct>
+          |    <item>
+          |      <field1>
+          |        <item>
+          |          <inner1>str1</inner1>
+          |        </item>
+          |        <item>
+          |          <inner2><item>str2</item><item>str22</item></inner2>
+          |        </item>
+          |      </field1>
+          |      <field2>
+          |        <array><item>1</item><item>2</item></array>
+          |        <array><item>3</item><item>4</item></array>
+          |      </field2>
+          |    </item>
+          |    <item>
+          |      <field1>
+          |        <item>
+          |          <inner2>
+          |            <item>str3</item><item>str33</item>
+          |          </inner2>
+          |        </item>
+          |        <item>
+          |          <inner1>str4</inner1>
+          |        </item>
+          |      </field1>
+          |      <field2>
+          |        <array>
+          |          <item>5</item><item>6</item>
+          |        </array>
+          |        <array>
+          |          <item>7</item><item>8</item>
+          |        </array>
+          |      </field2>
+          |    </item>
+          |  </complexArrayOfStruct>
+          |  <arrayOfArray1>
+          |  <array>
+          |    <item>5</item>
+          |  </array>
+          |</arrayOfArray1>
+          |<arrayOfArray1>
+          |  <array>
+          |    <item>6</item><item>7</item>
+          |  </array>
+          |  <array>
+          |    <item>8</item>
+          |  </array>
+          |</arrayOfArray1>
+          |  <arrayOfArray2>
+          |  <array>
+          |    <item>
+          |      <inner1>str1</inner1>
+          |    </item>
+          |  </array>
+          |</arrayOfArray2>
+          |<arrayOfArray2>
+          |  <array/>
+          |  <array>
+          |    <item>
+          |      <inner2>str3</inner2>
+          |      <inner2>str33</inner2>
+          |    </item>
+          |    <item>
+          |      <inner2>str4</inner2>
+          |      <inner1>str11</inner1>
+          |    </item>
+          |  </array>
+          |</arrayOfArray2>
+          |<arrayOfArray2>
+          |  <array>
+          |    <item>
+          |      <inner3>
+          |        <inner4>2</inner4>
+          |        <inner4>3</inner4>
+          |      </inner3>
+          |      <inner3/>
+          |    </item>
+          |  </array>
+          |</arrayOfArray2>
+          |</ROW>
+          |""".stripMargin :: Nil
+
+  def nullsInArrays: Seq[String] =
```

Comment:
No test for this?

Reply:
I plan to add tests in a follow-up PR. I will remove this for clarity in this PR.